spec/unit/box_spec.rb

```diff
@@ -0,0 +1,150 @@
+include BrowserConfigHelper
+
+describe BrowseEverything::Driver::Box, vcr: { cassette_name: 'box', record: :none } do
+  before(:all)  { stub_configuration   }
+  after(:all)   { unstub_configuration }
+
+  let(:browser) { BrowseEverything::Browser.new(url_options) }
+  let(:provider) { browser.providers['box'] }
+  let(:auth_params) do
+    {
+      'code' => 'CODE',
+      'state' => 'box'
+    }
+  end
+
+  # To re-record the VCR responses, you will need to substitute these values with the actual tokens returned from Box.
+  let(:token) do
+    {
+      'token' => 'TOKEN',
+      'refresh_token' => 'REFRESH_TOKEN'
+    }
+  end
+
+  subject    { provider }
+
+  its(:name) { is_expected.to eq('Box') }
+  its(:key)  { is_expected.to eq('box') }
+  its(:icon) { is_expected.to be_a(String) }
+
+  describe '#validate_config' do
+    it 'raises and error with an incomplete configuration' do
+      expect { BrowseEverything::Driver::Box.new({}) }.to raise_error(BrowseEverything::InitializationError)
+    end
+  end
+
+  describe '#auth_link' do
+    subject { provider.auth_link }
+
+    it { is_expected.to start_with('https://www.box.com/api/oauth2/authorize') }
+    it { is_expected.to include('browse%2Fconnect') }
+    it { is_expected.to include('response_type') }
+  end
+
+  describe '#authorized?' do
+    subject { provider.authorized? }
+    context 'when the access token is not registered' do
+      it { is_expected.to be(false) }
+    end
+    context 'when the access tokens are registered and not expired' do
+      before { provider.token = token.merge('expires_at' => Time.now.to_i + 360) }
+      it { is_expected.to be(true) }
+    end
+    context 'when the access tokens are registered but no expiration time' do
+      before { provider.token = token }
+      it { is_expected.to be(false) }
+    end
+    context 'when the access tokens are registered but expired' do
+      before { provider.token = token.merge('expires_at' => Time.now.to_i - 360) }
+      it { is_expected.to be(false) }
+    end
+  end
+
+  describe '#connect' do
+    it 'registers new tokens' do
+      expect(provider).to receive(:register_access_token).with(kind_of(OAuth2::AccessToken))
+      provider.connect(auth_params, 'data')
+    end
+  end
+
+  describe '#contents' do
+    before { provider.token = token }
+
+    context 'of the root directory' do
+      let(:contents) { provider.contents('') }
+
+      describe 'the first item' do
+        subject { contents[0] }
```

Comment:
This contents [0] is a bit magical.  I wonder if a comment here to point to the VCR tape may make it less so.